### ISSUE TITLE
ADA - Buttons missing label and expand/collapse notification

### ISSUE THREAD

**Issue by artjynx** [TAMULib/tamu-library-components#423]
The buttons highlighted in image below are missing label and submenu items not functional with a keyboard only tab key. Only mouse hover is able to expand the dropdown.
![image](https://user-images.githubusercontent.com/1204813/186003308-bca7b694-8f7b-4574-a373-268dc654ca28.png)

Navigation buttons missing label
Collapsible content areas are presented in websites and applications as a way to let users control how content is shown or hidden on the page. The navigation dropdown needs to be operable by a keyboard with status of expanded or collapsed notified to assistive technology, especially a screen reader. There is no notification to the user about the dropdown items.
Accessibility guidelines
[2.1.1 Keyboard (Level A)](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.w3.org%2FWAI%2FWCAG21%2FUnderstanding%2Fkeyboard.html&data=05%7C01%7Cartjynx%40library.tamu.edu%7Cabb64fa5c8244c6c447e08da64cbe4ac%7C68f381e346da47b9ba576f322b8f0da1%7C1%7C0%7C637933124123818269%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=MScoUb88ZpvoR2tUg8HNt%2B0bHiPQJq97YoKdox%2BagaM%3D&reserved=0): All functionality of the content is operable through a keyboard interface without requiring specific timings for individual keystrokes, except where the underlying function requires input that depends on the path of the user's movement and not just the endpoints.
[4.1.2 Name, Role, Value](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.w3.org%2FTR%2FUNDERSTANDING-WCAG20%2Fensure-compat-rsv.html&data=05%7C01%7Cartjynx%40library.tamu.edu%7Cabb64fa5c8244c6c447e08da64cbe4ac%7C68f381e346da47b9ba576f322b8f0da1%7C1%7C0%7C637933124123818269%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=JA1UQpPkgcUZU35cMXTxOZbkZcp%2ByhgWwJaNxEjDA7M%3D&reserved=0): For all user interface components (including but not limited to: form elements, links and components generated by scripts), the name and role can be programmatically determined; states, properties, and values that can be set by the user can be programmatically set; and notification of changes to these items is available to user agents, including assistive technologies. (Level A)
Recommendation
Use an accordion for the buttons highlighted above to ensure the expand/collapse region is also notified to screen reader. Keyboard functionality needs to be added through JavaScript. Visit [Accordion Example snippet](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.w3.org%2FWAI%2FARIA%2Fapg%2Fexample-index%2Faccordion%2Faccordion&data=05%7C01%7Cartjynx%40library.tamu.edu%7Cabb64fa5c8244c6c447e08da64cbe4ac%7C68f381e346da47b9ba576f322b8f0da1%7C1%7C0%7C637933124123818269%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=P%2FrfYn0RY4Lrpyo7VNzkj69hpE28JRpMdLglMqyLfyY%3D&reserved=0) to resolve the issue.

**Comment by jcreel** [comment]
One potential course of action:  Change the buttons to links so keyboard users can avoid the dropdowns.

**Comment by kaladay** [comment]
For reference, this is a more official representation on how to fully perform the expand/collapsed behavior:
- https://www.w3.org/WAI/ARIA/apg/example-index/accordion/accordion.html

It includes using `aria-controls` attribute for designating what is being controlled, `hidden` attribute for formally hiding something, and even uses `role="region".

The one problem I see is that they use `<fieldset>` which is known to not work in all environments, such as when printing.
So for people who use printed versions of pages, it becomes less accessible to them when a `<fieldset>` is used.
